src/v/kafka/server/handlers/describe_log_dirs.cc

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+#include "kafka/server/handlers/describe_log_dirs.h"
+
+#include "cluster/partition_manager.h"
+#include "kafka/protocol/errors.h"
+#include "kafka/server/request_context.h"
+#include "kafka/server/response.h"
+#include "model/fundamental.h"
+#include "model/namespace.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/smp.hh>
+#include <seastar/util/log.hh>
+
+#include <fmt/ostream.h>
+
+namespace kafka {
+
+using part_set
+  = absl::flat_hash_map<model::topic, std::vector<describe_log_dirs_partition>>;
+
+static describe_log_dirs_partition describe_partition(cluster::partition& p) {
+    return describe_log_dirs_partition{
+      .partition_index = p.ntp().tp.partition(),
+      .partition_size = static_cast<int64_t>(p.size_bytes()),
+      .offset_lag = 0,
```

Comment:
Would be good to leave a comment here.

Reply:
according to [this](https://stackoverflow.com/questions/39203215/kafka-difference-between-log-end-offsetleo-vs-high-watermarkhw) LEO may be higher that highwatermark and in our case it seems it's the difference between `_visibility_upper_bound_index` (I think it's the same as `dirty_offset`) and `_majority_replicated_index` (majority of dirty offsets) //cc @mmaslankaprv